packages/teams-js/test/public/mail.spec.ts

```diff
@@ -0,0 +1,325 @@
+        .openMailItem(openMailItemParams)
+        .catch(e => expect(e).toMatchObject(new Error('The library has not yet been initialized')));
+    });
+
+    it('should not allow calls from settings context', async () => {
+      await utils.initializeWithContext('settings');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "settings".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from authentication context', async () => {
+      await utils.initializeWithContext('authentication');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error(
+              'This call is only allowed in following contexts: ["content"]. Current context: "authentication".',
+            ),
+          ),
+        );
+    });
+
+    it('should not allow calls from remove context', async () => {
+      await utils.initializeWithContext('remove');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "remove".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from task context', async () => {
+      await utils.initializeWithContext('task');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "task".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from sidePanel context', async () => {
+      await utils.initializeWithContext('sidePanel');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "sidePanel".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from stage context', async () => {
+      await utils.initializeWithContext('stage');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "stage".'),
+          ),
+        );
+    });
+
+    it('should not allow calls from meetingStage context', async () => {
+      await utils.initializeWithContext('meetingStage');
+
+      await mail
+        .openMailItem(openMailItemParams)
+        .catch(e =>
+          expect(e).toMatchObject(
+            new Error('This call is only allowed in following contexts: ["content"]. Current context: "meetingStage".'),
+          ),
+        );
+    });
```

Comment:
It feels like all these tests should be a loop over the list of possible contexts or something. Or at least a one-line call to a helper method that takes the context as a parameter?

Reply:
No problem